### ISSUE TITLE
Add missing data to datasets

### ISSUE THREAD

**Issue by mschoettle** [jeremymoreau/covid19mtl#30]
Data is missing from September 3rd onwards.

I haven't figured out yet how the INSPQ data can be replaced (due to the removal of the old CSV files). If someone can tell me which data exactly is used from them on the dashboard that would be great :)

**Comment by mschoettle** [comment]
There is fortunately this archive where we should be able to get all the missing data from: https://github.com/jeanpaulrsoucy/covid-19-canada-gov-data

The data format from Santé Montreal (`municipal.csv`) changed on Oct 9. The column "Nombre de cas confirmés" (target index `0`) is now "Nombre de cas cumulatif, depuis le début de la pandémie" (target index `3`).

**Comment by mschoettle** [comment]
Santé Montreal unfortunately only updates the data on their website during the week. Even today (Sunday), the data seems to be shown from Thursday ("Source : Fichier Trajectoire de santé publique (TSP), extracted on 19 November, 5:29 PM") while on the top of the tables it states "Data as of November 20, 2020, at 12 p.m.").

Anyway, I see that sometimes the `cases.csv` contains columns filled with `na`. How was this handled when adding data?

**Comment by mschoettle** [comment]
INSPQ has the following new CSV files:

* `https://www.inspq.qc.ca/sites/default/files/covid/donnees/covid19-hist.csv`: If I see it correctly, this would be the replacement for `combine.csv`, however, the data format changed
* `https://www.inspq.qc.ca/sites/default/files/covid/donnees/tableau-rls-new.csv`: Data for the different region's RLS
* `https://www.inspq.qc.ca/sites/default/files/covid/donnees/tableau-rpa-new.csv`: Data for the deaths by place. This file has a change in columns, although this happened even earlier (the column `Inconnue` was dropped, currently it is together with `Domicile` as `Domicile et Inconnu`)

There are also the following files available:

* `https://www.inspq.qc.ca/sites/default/files/covid/donnees/regions.csv`
* `https://www.inspq.qc.ca/sites/default/files/covid/donnees/manual-data.csv`

**Comment by jeremymoreau** [comment]
The "Regroupement" and "Croisement" columns are key in the new covid19-hist.csv  file. For all the variables below we need to filter out all the rows except those where Regroupement == "Sexe" and Croisement == "TOT"
The other rows might be interesting also, but aren't currently used in the dashboard for the QC data.

Here's the correspondence between the column names used in data_qc.csv and the covid19-hist.csv column names:
`variable name parsed by covid19mtl : covid19-hist.csv column name`
```
date : Date

cases_qc : cas_cum_tot_n
new_cases_qc : cas_quo_tot_n
active_cases : act_cum_tot_n

deaths_qc : dec_cum_tot_n
new_deaths_qc : dec_quo_tot_n

negative_tests_qc : psi_cum_inf_n
```
We should probably use psi_cum_pos_n for the positive tests count in the "Diagnostics tests (QC)" graph instead of just plotting cases_qc/cas_cum_tot_n again (since that number includes epidemiologically linked cases that weren't tested)


Hospitalisation numbers in covid19-hist.csv are either cumulative or new cases (but not active active cases, which we want)

Would need to find/scrape the data used in fig. 3.1 here: https://www.inspq.qc.ca/covid-19/donnees
```
hospitalisations_all : 
icu_qc : 
```


Also found this repo that has a lot of useful info about the covid19-hist.csv file (also a cool project!): https://github.com/SimonCoulombe/covidtwitterbot

**Comment by mschoettle** [comment]
Thanks, that's really helpful. 

The same data like in `Sexe` & `TOT` is also present in `Région` & `RS99` (Nom == `Ensemble du Québec`). There's one difference between the two rows I noticed, which is the test positivity. In the case of "Sexe/TOT" the calculation seems off whereas for "Ensemble du Québec" it is == `psi_quo_pos_n`/`psi_quo_tes_n`

If I see this right we can't just append rows from the new `covid19-hist.csv` to `processed/data_qc.csv`. `data_qc.csv` contains all the other columns that are unused and were not renamed but `covid19-hist.csv` contains others.

I would therefore suggest to recreate it from scratch since `covid19-hist.csv` contains past data.

> We should probably use psi_cum_pos_n for the positive tests count in the "Diagnostics tests (QC)" graph instead of just plotting cases_qc/cas_cum_tot_n again (since that number includes epidemiologically linked cases that weren't tested)

I agree, plus the positivity rate :). The unfortunate thing is that those numbers are delayed by a day.

> Hospitalisation numbers in covid19-hist.csv are either cumulative or new cases (but not active active cases, which we want)

I guess since there are no discharges in there we can't even calculate the active hospitalisations.
According to this cool Twitter bot repo you found, the old hospitalisation numbers are in the `manual-data.csv`. Another option, although more difficult, is to automatically save the chart as a CSV (there's a menu option to do that). But that requires something like Selenium to automate.

**Comment by mschoettle** [comment]
Fixed with #33.
